### PR TITLE
feat: Alias `spew.Dump` and so on

### PR DESCRIPTION
**Pull request by flc1125**
## 📑 Description

Alias `spew.Dump` and so on.

- `debug.Dump()`
- `debug.SDump()`
- `debug.FDump()`

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] All the tests have passed

### CODE REVIEW

**Comment by DevHaoZi** [comment]
I checked the package `github.com/davecgh/go-spew/spew`, I don't think its implementation is complicated, so I suggest implementing your own instead of importing a 3rd party package.

**Comment by flc1125** [comment]
I think we can make a set of tools about Debug, so that when using the framework, it can be very convenient to use it.

This PR, however, is only part of a collection of debug tools. It already has `spew`, but we're just referencing it and putting it together.

While we can develop a set of features that are the same as `spew`, why repeat the wheel when we have open source?

These are my thoughts and I'd like to hear yours.

**Comment by DevHaoZi** [comment]
> I think we can make a set of tools about Debug, so that when using the framework, it can be very convenient to use it.
> 
> This PR, however, is only part of a collection of debug tools. It already has `spew`, but we're just referencing it and putting it together.
> 
> While we can develop a set of features that are the same as `spew`, why repeat the wheel when we have open source?
> 
> These are my thoughts and I'd like to hear yours.

1. This package lacks maintenance, last updated in 2018.
2. We are trying to minimize the use of unnecessary third-party packages. Refer to Laravel's github.com/illuminate repository, which uses very few third-party packages.

File: support/debug/dump_test.go
Comment (hwbrzzl) [inline support/debug/dump_test.go:1]: S: I think the unit tests are unnecessary, they're so simple.

File: support/debug/dump.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package debug
+
+import (
+	"github.com/davecgh/go-spew/spew"
+	"io"
+)
+
+// Dump is a wrapper around spew.Dump.
```
Comment (hwbrzzl) [inline support/debug/dump.go:8]: S: Add the real comment here.

File: support/debug/dump.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+func Dump(v ...interface{}) {
+	spew.Dump(v...)
+}
+
+// FDump is a wrapper around spew.Fdump.
```
Comment (hwbrzzl) [inline support/debug/dump.go:13]: Ditto

File: support/debug/dump.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+func FDump(w io.Writer, v ...interface{}) {
+	spew.Fdump(w, v...)
+}
+
+// SDump is a wrapper around spew.Sdump.
```
Comment (hwbrzzl) [inline support/debug/dump.go:18]: Ditto

**Review by hwbrzzl — COMMENTED** [review]
Good feature! The meaning of @DevHaoZi is want to reduce dependencies, that's a good idea, but I think we can use the third package in this PR directly, because there is some complex logic and the package is small.

File: support/debug/dump.go (hunk 2 above)
Comment (flc1125) [reply support/debug/dump.go]: Optimized

File: support/debug/dump.go (hunk 1 above)
Comment (flc1125) [reply support/debug/dump.go]: Optimized

File: support/debug/dump_test.go
Comment (flc1125) [reply support/debug/dump_test.go:1]: Optimized

**Comment by hwbrzzl** [comment]
> > I think we can make a set of tools about Debug, so that when using the framework, it can be very convenient to use it.
> > This PR, however, is only part of a collection of debug tools. It already has `spew`, but we're just referencing it and putting it together.
> > While we can develop a set of features that are the same as `spew`, why repeat the wheel when we have open source?
> > These are my thoughts and I'd like to hear yours.
> 
> 1. This package lacks maintenance, last updated in 2018.
> 2. We are trying to minimize the use of unnecessary third-party packages. Refer to Laravel's `github.com/illuminate` repository, which uses very few third-party packages.

Sorry, I didn't notice that this package has not been maintained for a long time, this's indeed a problem. Can we find another package instead? @flc1125

**Comment by flc1125** [comment]
> > > I think we can make a set of tools about Debug, so that when using the framework, it can be very convenient to use it.
> > > This PR, however, is only part of a collection of debug tools. It already has `spew`, but we're just referencing it and putting it together.
> > > While we can develop a set of features that are the same as `spew`, why repeat the wheel when we have open source?
> > > These are my thoughts and I'd like to hear yours.
> > 
> > 
> > 
> > 1. This package lacks maintenance, last updated in 2018.
> > 2. We are trying to minimize the use of unnecessary third-party packages. Refer to Laravel's `github.com/illuminate` repository, which uses very few third-party packages.
> 
> Sorry, I didn't notice that this package has not been maintained for a long time, this's indeed a problem. Can we find another package instead? @flc1125

I can try to find out.

As far as I know, the spew package is used in many projects. The framework itself also relies on the spew package.

**Comment by DevHaoZi** [comment]
> > > > I think we can make a set of tools about Debug, so that when using the framework, it can be very convenient to use it.
> > > > This PR, however, is only part of a collection of debug tools. It already has `spew`, but we're just referencing it and putting it together.
> > > > While we can develop a set of features that are the same as `spew`, why repeat the wheel when we have open source?
> > > > These are my thoughts and I'd like to hear yours.
> > > 
> > > 
> > > 
> > > 1. This package lacks maintenance, last updated in 2018.
> > > 2. We are trying to minimize the use of unnecessary third-party packages. Refer to Laravel's `github.com/illuminate` repository, which uses very few third-party packages.
> > 
> > 
> > Sorry, I didn't notice that this package has not been maintained for a long time, this's indeed a problem. Can we find another package instead? @flc1125
> 
> I can try to find out.
> 
> As far as I know, the spew package is used in many projects. The framework itself also relies on the spew package.

No, this framework doesn't relies spew package, maybe a 3rd package import it. That why we need to minimize the use of unnecessary 3rd packages.

**Comment by hwbrzzl** [comment]
@flc1125 Oh, good catch, it's true, `github.com/stretchr/testify/mock` is using it. Okay, that being the case, let's go ahead! What do you think about this, please?

**Comment by hwbrzzl** [comment]
@flc1125 Oh, good catch, it's true, github.com/stretchr/testify/mock is using it, we can't remove the `testify` package. Okay, that being the case, I'm ok with this! What do you think about this, please? @DevHaoZi

**Comment by DevHaoZi** [comment]
> 

OK, no problem.

File: support/debug/dump_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func redirectStdout(fn func()) ([]byte, error) {
```
Comment (hwbrzzl) [inline support/debug/dump_test.go:11]: Good!

**Comment by flc1125** [comment]
My original intention was to facilitate debugging, and spew is a good and relatively mature third-party package in this area.

My proposal can be considered as follows:

- Does our project need to explore the debug area in the near future?
- Is our project looking to reduce reliance on third party packages in the near future?

I think we can choose from the above direction.

----

- If the final conclusion is to use, then I am deeply honored. Even in future planning, remove this code.
- If the final conclusion is not to use, I think it is also a good decision. 
	
	> The PR also only defines aliases, and if the user wishes to introduce spew, the option of this package can be given to the user.

In your way, choose your correct answer.

---

As I was writing the above, I was honored to find that you had merged. Thank.

@hwbrzzl @DevHaoZi 😁

**Comment by hwbrzzl** [comment]
It's good for goravel to have the feature, thanks for your contribution, hope you can become a core contributor in the future. 👍

**Comment by DevHaoZi** [comment]
> My original intention was to facilitate debugging, and spew is a good and relatively mature third-party package in this area.
> 
> My proposal can be considered as follows:
> 
> * Does our project need to explore the debug area in the near future?
> * Is our project looking to reduce reliance on third party packages in the near future?
> 
> I think we can choose from the above direction.
> 
> * If the final conclusion is to use, then I am deeply honored. Even in future planning, remove this code.
> * If the final conclusion is not to use, I think it is also a good decision.
>   > The PR also only defines aliases, and if the user wishes to introduce spew, the option of this package can be given to the user.
> 
> In your way, choose your correct answer.
> 
> As I was writing the above, I was honored to find that you had merged. Thank.
> 
> @hwbrzzl @DevHaoZi 😁

Yes, we are working on minimize the use of unnecessary 3rd packages. Before the next contribution, you can discuss it on github.com/goravel/goravel/issues, which can reduce unnecessary conflicts and troubles.😁

**Comment by flc1125** [comment]
> > My original intention was to facilitate debugging, and spew is a good and relatively mature third-party package in this area.
> > My proposal can be considered as follows:
> > 
> > * Does our project need to explore the debug area in the near future?
> > * Is our project looking to reduce reliance on third party packages in the near future?
> > 
> > I think we can choose from the above direction.
> > 
> > * If the final conclusion is to use, then I am deeply honored. Even in future planning, remove this code.
> > * If the final conclusion is not to use, I think it is also a good decision.
> >   > The PR also only defines aliases, and if the user wishes to introduce spew, the option of this package can be given to the user.
> > 
> > In your way, choose your correct answer.
> > As I was writing the above, I was honored to find that you had merged. Thank.
> > @hwbrzzl @DevHaoZi 😁
> 
> Yes, we are working on minimize the use of unnecessary 3rd packages. Before the next contribution, you can discuss it on github.com/goravel/goravel/issues first, which can reduce unnecessary conflicts and troubles.😁

Very good advice.

**Pull request merged**